### ISSUE TITLE
Per-Jump Encounter/Event Flags

### ISSUE THREAD

**Issue by puppetsquid** [Niels-NTG/FTLAV#6]
I posted about this in the FTL forum thread under the username 'Gencool', but am now posting here for ease of use.

I know "Crew skills as difference per sector" has already been added to the roadmap, but I think 'per Jump' would also be really useful.

The main example, again, is 'Crew Size' - just having a difference per sector means a ship might display having gained 1 crew over the sector, when in reality the player gained 5 and lost 4.

I suggested a [rough workaround](http://www.ftlgame.com/forum/viewtopic.php?f=7&t=28728&p=97104#p97104) for finding more in-depth information about crew deaths. 
I admit it becomes less accurate if more than one crewmember dies between jumps, and could probably be improved upon who knows how to code properly, but it'd be nice to see the crew data visualised in more detail.

With the data available through the FTL profile Editor, it seems this could potentially also be used to show Hazard data, Ships Destroyed/Surrendered/Escaped, Flagship Battles, Clones Created, Stores Visited, Systems Purchased/Upgraded, Augments/Weapons/Drones Added to Cargo/Equipped/Unequipped and even Event/Enemy Ship Names (where available). 
Having them as Per-Jump flags, rather than (or as well as) sector breakdowns could also help correlate them with dips in hull or peaks in scrap...

Some of this would probably have to be roll-over data, and all optional, though it'd be -amazing- if you could add mini weapon/ship/crew sprites later on (dug from game files, of course).

**Comment by Niels-NTG** [comment]
You're right about the difference per beacon thing. I decided to scrap the difference per sector feature entirely, since they are functionally the same idea. Only getting a difference per beacon offers a lot more valuable information, as you rightly pointed out.

As for your rough workaround... I will address it on a later date when I have more free time.

Most of the things you mentioned are already implemented in the CSV export feature. They haven't been added to the graph yet. I'm planning to add mouse-over pop-ups with qualitative data in a future version of FTLAV.

It's hard to really get a distinction between ships that have surrendered, destroyed or escaped. And getting event text is easy, but getting the choices is not. Enemy ship names are retrievable, but from experiments I can tell you it's not that interesting. They are often the same and not ment to be seen by the player.

Adding tiny sprites directly from the game files is something you should definitely expect to see in the future.

BTW: Next time you file an issue anywhere on GitHub, try to break it down in multiple separate issues so developers can tackle it more easily.

**Comment by Niels-NTG** [comment]
FTLAV v3.0.0 just got released. The export function does record more detailed encounter information than the previous versions of FTLAV did.

However, it isn't quite 100% reliable yet due to the FTL save game file format being mysterious at times. Often the only time a save game file can be parsed successfully is when the enemy ship has been defeated or escaped, meaning that the beacon gets marked as having no hostile ships, which isn't actually reflective of what just happened in-game.

I would like to address this issue in a future version of FTLAV, which hopefully won't take me another 9 years… :sweat_smile:

**Comment by Niels-NTG** [comment]
[FTLAV 3.1.0](https://github.com/Niels-NTG/FTLAV/releases/tag/v3.1.0) introduces per-beacon event markers.
